pkg/kernel/kernel.go

```diff
@@ -182,6 +203,17 @@ func (k *kernel) Run() {
 		}
 
 		k.waitStopped()
+
+		hasErr := false
+		for _, stage := range k.stages {
+			if stage.err != nil && stage.err != ErrKernelStopping {
```

Comment:
i wonder if it is ok for some modules to now properly shutdown.
Let's imagine there are metrics that weren't flushed to e.g. cloudwatch.
While the application did a great job and everything was good, we would still return an error code because metrics weren't flushed.
This is something we might want to know about, however we'd never check the exit code of an app for that - would we?
I think, we should only consider the application specific modules and not the whole module universe here.

Reply:
For now would I like to start with the whole module universe..  
But in general you could be right, that some "debugging" modules should not affect the exit code..